src/decide/Decider.java

```diff
@@ -93,7 +95,41 @@ public boolean lic3(){
     }
 
     public boolean lic4(){
-        return true;
+        // false in case of invalid parameters
+        if(this.parameters.Q_PTS < 2 || this.parameters.Q_PTS > numpoints){
+            return false;
```

Comment:
Missing method description.

Reply:
It's added in the commit for that issue, this was just branched from that branch so not completely accurate.